API/src/main/java/de/brandwatch/minianalytics/api/service/MentionService.java

```diff
@@ -0,0 +1,94 @@
+package de.brandwatch.minianalytics.api.service;
+
+import com.google.gson.Gson;
+import de.brandwatch.minianalytics.api.postgres.repository.QueryRepository;
+import de.brandwatch.minianalytics.api.security.model.User;
+import de.brandwatch.minianalytics.api.security.repositories.UserRepository;
+import de.brandwatch.minianalytics.api.solr.model.Mention;
+import de.brandwatch.minianalytics.api.solr.repository.MentionRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.List;
+
+@Component
+public class MentionService {
+
+    private static final Logger logger = LoggerFactory.getLogger(MentionService.class);
+
+    private final MentionRepository mentionRepository;
+
+    private final UserRepository userRepository;
+
+    private final QueryRepository queryRepository;
+
+    private static final Gson gson = new Gson();
+
+    @Autowired
+    public MentionService(MentionRepository mentionRepository, UserRepository userRepository, QueryRepository queryRepository) {
+        this.mentionRepository = mentionRepository;
+        this.userRepository = userRepository;
+        this.queryRepository = queryRepository;
+    }
+
+    public List<Mention> getMentionsFromQueryID(String queryID, String startDate, String endDate) throws Exception {
+
+        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        String username = ((UserDetails) principal).getUsername();
+
+        User user = userRepository.findByUsername(username);
+
+        if(!queryRepository.findById(Long.valueOf(queryID)).get().getUser().equals(user)){
+            throw new Exception("Query " + queryID + " doesnt belong to user " + username);
+        }
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("[");
+
+        if(!endDate.equals("") && !startDate.equals("")){
```

Comment:
Not sure if you need to check these two dates are empty or null. For the empty case, better be :
`!"".equals(endDate) && !"".equals(startDate)`

Reply:
Isn't `!endDate.equals("")` and `!"".equals(endDate)` the exact same? Just asking for clarification.